tests/CriticalBackgroundServiceTests.cs

```diff
@@ -42,32 +43,13 @@ public void WhenBackgroundService_ErrorsWithoutYielding_ApplicationCrashes()
         [Fact]
         public async Task WhenCriticalBackgroundService_YieldsBeforeThrowingError_ApplicationShouldCrash()
         {
-            var applicationEnder = new ApplicationEnderMock();
+            var applicationEnder = new ApplicationEnderTaskMock();
 
-            var factory = this._factory.WithWebHostBuilder(builder =>
-            {
-                builder.ConfigureTestServices(services =>
-                {
-                    services.AddTransient<IApplicationEnder>(s => applicationEnder);
-                    services.AddHostedService<YieldingAndThenCrashingCriticalBackgroundService>();
-                });
-            });
-
-            var client = factory.CreateClient();
-            var res = await client.GetAsync("/");
-
-            // Task is hella flaky because it depends on the internals of the IHostedService - try yielding a bunch of times
-            // to hope that it's done requesting application shutdown at this point
-            for (int i = 0; i < 10; i++)
-            {
-                await Task.Delay(50);
-                await Task.Yield();
-            }
-
-            // due to https://github.com/dotnet/aspnetcore/issues/25857 we can't test if the process is closed directly
-            applicationEnder.ShutDownRequested.Should().BeTrue();
+            YieldingAndThenCrashingCriticalBackgroundService backgroundService = new(applicationEnder);
 
+            await backgroundService.StartAsync(CancellationToken.None);
 
+            Task.WaitAny(new Task[] { applicationEnder.ShutDownTask }, 3000).Should().Be(0);
```

Comment:
Hm, this test doesn't test quite as well the real-life behaviour of the backgorundservice, however it has the advantage of actually passing, so let's roll with that. Good job!

Reply:
Yes, I know. I think, unit testing is more useful than integration testing in this case

P.S. For example, I use background service without any web context